plugins/woocommerce-admin/client/products/fields/attribute-field/add-attribute-modal.tsx

```diff
@@ -0,0 +1,295 @@
+/**
+ * External dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { Button, Modal, Notice } from '@wordpress/components';
+import { trash } from '@wordpress/icons';
+import { ProductAttribute, ProductAttributeTerm } from '@woocommerce/data';
+import { Form } from '@woocommerce/components';
+
+/**
+ * Internal dependencies
+ */
+import './add-attribute-modal.scss';
+import { AttributeInputField } from '../attribute-input-field';
+import { AttributeTermInputField } from '../attribute-term-input-field';
+
+type CreateCategoryModalProps = {
+	onCancel: () => void;
+	onAdd: ( newCategories: ProductAttribute[] ) => void;
+	selectedAttributeIds?: number[];
+};
+
+type AttributeForm = {
+	attributes: {
+		attribute?: ProductAttribute;
+		terms: ProductAttributeTerm[];
+	}[];
+};
+
+export const AddAttributeModal: React.FC< CreateCategoryModalProps > = ( {
+	onCancel,
+	onAdd,
+	selectedAttributeIds = [],
+} ) => {
+	const addAnother = (
+		values: AttributeForm,
+		setValue: (
+			name: string,
+			value: AttributeForm[ keyof AttributeForm ]
+		) => void
+	) => {
+		setValue( 'attributes', [
+			...values.attributes,
+			{
+				attribute: undefined,
+				terms: [],
+			},
+		] );
+	};
+
+	const onAddingAttributes = ( values: AttributeForm ) => {
+		const newAttributesToAdd: ProductAttribute[] = [];
+		values.attributes.forEach( ( attr ) => {
+			if (
+				attr.attribute &&
+				attr.attribute.name &&
+				attr.terms.length > 0
+			) {
+				newAttributesToAdd.push( {
+					...( attr.attribute as ProductAttribute ),
+					options: attr.terms.map( ( term ) => term.name ),
+				} );
+			}
+		} );
+		onAdd( newAttributesToAdd );
+	};
+
+	const onRemove = (
+		index: number,
+		values: AttributeForm,
+		setValue: (
+			name: string,
+			value: AttributeForm[ keyof AttributeForm ]
+		) => void
+	) => {
+		if ( values.attributes.length > 1 ) {
+			setValue(
+				'attributes',
+				values.attributes.filter( ( val, i ) => i !== index )
+			);
+		} else {
+			setValue( `attributes[${ index }]`, [
+				{ attribute: undefined, terms: [] },
+			] );
+		}
+	};
+
+	const focusValueField = ( index: number ) => {
+		const valueInputField: HTMLInputElement | null = document.querySelector(
+			'.woocommerce-add-attribute-modal__table-row-' +
+				index +
+				' .woocommerce-add-attribute-modal__table-attribute-value-column .woocommerce-experimental-select-control__input'
+		);
+		if ( valueInputField ) {
+			setTimeout( () => {
+				valueInputField.focus();
+			}, 0 );
+		}
+	};
+
+	return (
+		<Modal
+			title={ __( 'Add attributes', 'woocommerce' ) }
+			onRequestClose={ () => onCancel() }
```

Comment:
I wonder if we might want a conditional confirmation dialogue here if they've made changes, since the user will lose any attributes they've plugged in on accidental clicks.

Reply:
Yeah that is a good suggestion, I can definitely add this. I could add a browser alert, similar to the one we show when deleting on of the attributes?
cc'ing @jarekmorawski on this.

Reply:
I did add this btw as part of: https://github.com/woocommerce/woocommerce/pull/34999/commits/3f269cf405508127b038b13a71a1a50a6fa7b2df
Let me know if the copy should change?
<img width="1033" alt="Screen Shot 2022-10-19 at 2 20 44 PM" src="https://user-images.githubusercontent.com/2240960/196760813-d4511fc1-a499-4b0c-a7b7-72f77917e974.png">